src/mime.cc

```diff
@@ -262,70 +259,73 @@ mimeInit(char *filename)
     mimeFreeMemory();
 
     while (fgets(buf, BUFSIZ, fp)) {
-        if ((t = strchr(buf, '#')))
-            *t = '\0';
 
-        if ((t = strchr(buf, '\r')))
-            *t = '\0';
+        try {
 
-        if ((t = strchr(buf, '\n')))
-            *t = '\0';
+            if ((t = strchr(buf, '#')))
+                *t = '\0';
 
-        if (buf[0] == '\0')
-            continue;
+            if ((t = strchr(buf, '\r')))
+                *t = '\0';
 
-        xstrncpy(chopbuf, buf, BUFSIZ);
+            if ((t = strchr(buf, '\n')))
+                *t = '\0';
 
-        if ((pattern = strtok(chopbuf, w_space)) == nullptr) {
-            debugs(25, DBG_IMPORTANT, "ERROR: mimeInit: parse failure: '" << buf << "'");
-            continue;
-        }
+            if (buf[0] == '\0')
+                continue;
 
-        if ((type = strtok(nullptr, w_space)) == nullptr) {
-            debugs(25, DBG_IMPORTANT, "ERROR: mimeInit: parse failure: '" << buf << "'");
-            continue;
-        }
+            xstrncpy(chopbuf, buf, BUFSIZ);
 
-        if ((icon = strtok(nullptr, w_space)) == nullptr) {
-            debugs(25, DBG_IMPORTANT, "ERROR: mimeInit: parse failure: '" << buf << "'");
-            continue;
-        }
+            const auto pattern = SBuf(strtok(chopbuf, w_space));
+            if (pattern.isEmpty()) {
```

Comment:
Do we have to change `pattern` type and initialization? It feels like an unrelated change.